### ISSUE TITLE
Freezes after deleting profile

### ISSUE THREAD

**Issue by tox-user** [qTox/qTox#4841]
##### Brief Description

OS: Debian 9
qTox version: v1.11.0-391-g531defd0 (the version is actually 1.13, but it shows 1.11)
Commit hash: 531defd0aa66af1b128f7293bc08718ce2cc064f
toxcore: 0.1.10
Qt: 5.7.1

Reproducible: Always

##### Steps to reproduce
1. Run qTox with -l option, then select or create a profile.
2. Go to profile settings and delete the profile.

##### Observed Behavior
After confirming the deletion UI becomes unresponsive, but qTox keeps running.

##### Expected Behavior
Go back to login screen instead of freezing.

##### Additional Info
```
persistence/profile.cpp:641 : Debug: Removing profile "test1"
persistence/db/rawdatabase.cpp:450 : Debug: Removing database  "/home/user/.config/tox/test1.db"
:0 : Warning: QMetaMethod::invoke: Dead lock detected in BlockingQueuedConnection: Receiver is Nexus(0x56187b2bbd60)
```

**Comment by Diadlo** [comment]
Can't reproduce

**Comment by sudden6** [comment]
closed, for reason see #4896
